### ISSUE TITLE
Fix S2857 FP: Rule is not checking SQL keywords in interepolated string with special characters

### ISSUE THREAD

**Issue by mbidois-izysolutions** [SonarSource/sonar-dotnet#8102]
### Description

Rule: S2857 fail to check for SQL keywords inside an interpolated string with special characters.
It may be redundant with issue #6633 but since this is not with a const string, i think it might be a different issue.

### Repro steps

```
using System.Data.SqlClient;

namespace TestSQFP;

public class S2857FP
{
	public void ExampleFP()
	{
		string one = "One";
		string myString = $"{{{one}}}"; // Noncompliant FP
	}
}
```

### Expected behavior

Rule to not react for this kind of string.

### Actual behavior

Rule reacts for this kind of string (and multiple times: before the inner string, and before the '}}')

### Known workarounds

Write the string differently (ie concat) or put the '{' directly in the string:
```
string oneWithBracket = "{One}";
string myStringWithBracket = $"{oneWithBracket}";
```

### Related information

* SonarLint for Visual Studio 2022, version 6.11.1.58935
* Visual Studio 2022
* dotnet 6
* Windows 11

**Comment by cristian-ambrosini-sonarsource** [comment]
Hi @mbidois-izysolutions! 
Yes, it's an FP similar to the issue you linked, however, the examples I provided were not enough to cover all cases, thanks for raising the point. I'll add more tests to document each of the following:

```cs
void Example(string parameter)
{
    const string constOne = "One";
    string nonConstOne = "One";
    string empty = string.Empty;

    const string s0 = $"{constOne}";            // Compliant
    const string s1 = $"{constOne}Two";         // Noncompliant FP
    const string s2 = $"{nameof(constOne)}Two"; // Noncompliant FP
    const string s3 = $"{parameter}";           // Error CS0133
    const string s4 = $"{parameter}Two";        // Error CS0133
    const string s5 = $"{nonConstOne}";         // Error CS0133
    const string s6 = $"{nonConstOne}Two";      // Error CS0133
                                                // Noncompliant@-1 FP
    const string s7 = $"{empty}";               // Error CS0133
    const string s8 = $"{empty}Two";            // Error CS0133

    string s9 = $"{constOne}";                  // Compliant
    string s10 = $"{constOne}Two";              // Noncompliant FP
    string s11 = $"{nameof(constOne)}Two";      // Noncompliant FP
    string s12 = $"{parameter}";                // Compliant
    string s13 = $"{parameter}Two";             // Compliant
    string s14 = $"{nonConstOne}";              // Compliant
    string s15 = $"{nonConstOne}Two";           // Noncompliant FP
    string s16 = $"{empty}";                    // Compliant
    string s17 = $"{empty}Two";                 // Compliant

    string s18 = $"{{{nonConstOne}}}";          // Noncompliant FP
                                                // Noncompliant@-1 FP
}
```

Let me know if I missed something, in the meantime I'll close this to centralize all the FP examples of this kind in the previous issue.

Thanks a lot
